### ISSUE TITLE
published image unpatched security vulnerabilities

### ISSUE THREAD

**Issue by artooro** [datastax/docker-images#13]
In the published image, the following security vulnerabilities were found by Google's image scan.

```
2 critical severity vulnerabilities
CVE-2017-13815 (on file 5.25 2ubuntu1)

CVE-2017-13846 (on pcre3 8.38 3.1)

15 high severity vulnerabilities
CVE-2017-8804 (on glibc 2.23 0ubuntu9)

CVE-2017-2518 (on sqlite3 3.11.0 1ubuntu1)

CVE-2017-2513 (on sqlite3 3.11.0 1ubuntu1)

CVE-2017-11462 (on krb5 1.13.2+dfsg 5ubuntu2)

CVE-2017-1000158 (on python3.5 3.5.2 2ubuntu0~16.04.3)

CVE-2017-10685 (on ncurses 6.0+20160213 1ubuntu1)

CVE-2017-7614 (on binutils 2.26.1 1ubuntu1~16.04.5)

CVE-2017-8283 (on dpkg 1.18.4ubuntu1.2 0)

CVE-2017-10684 (on ncurses 6.0+20160213 1ubuntu1)

CVE-2017-2520 (on sqlite3 3.11.0 1ubuntu1)

CVE-2017-2519 (on sqlite3 3.11.0 1ubuntu1)

CVE-2017-12424 (on shadow 4.2 3.1ubuntu5.3)

CVE-2016-2779 (on util-linux 2.27.1 6ubuntu3.3)

CVE-2016-4484 (on cryptsetup 1.6.6 5ubuntu2.1)

CVE-2016-1238 (on perl 5.22.1 9ubuntu0.2)
```

It seems like the image was built using an outdated version of ubuntu for some reason. The dse-base uses `ubuntu:16.04` and my assumption is that your build system has not been updating the base and upstream images to get the latest security patches.

**Comment by mshuler** [comment]
[base/Dockerfile#L13-L15](https://github.com/datastax/docker-images/blob/master/base/Dockerfile#L13-L15) might include an `apt-get dist-upgrade` to pull OS updates at local container creation time, then proceed with customization. PRs welcomed!

**Comment by artooro** [comment]
@mshuler it may be a better idea for the build script being used to pull the latest upstream image before building the base image.

**Comment by mshuler** [comment]
I'm no docker guru, but I assumed that is the action that `FROM ubuntu:16.04` performs. I thought we were talking about updates needed in the existing upstream image. Apologies for the misunderstanding.

**Comment by mshuler** [comment]
Since I came across it when digging a little, this appears to be a relatively well-known problem with little to no solution currently. As a bandaid, I suppose a pre-build step of deleting all local related docker images is the way other people handle this?
https://github.com/moby/moby/issues/13331

**Comment by NicoTexas** [comment]
Yes, this is a common pitfall and I think Docker is not doing a good job on this (latest tag is confusing for a lot of people).
My take is that using general tags like 'latest', or in that case '16.04' is an anti-pattern.
Your image should be built from an exact version, in that case
```
FROM ubuntu:xenial-20180123
```
